test/idle-source-e2e/testdata/kafka-pipeline.yaml

```diff
@@ -0,0 +1,57 @@
+apiVersion: numaflow.numaproj.io/v1alpha1
+kind: Pipeline
+metadata:
+  name: kafka-idle-source
+spec:
+  watermark:
+    idleSource:
+      threshold: 5s # The pipeline will be considered idle if the source has not emitted any data for given threshold value.
+      incrementBy: 3s # If source is found to be idle then increment the watermark by given incrementBy value.
+      stepInterval: 2s # If source is idling then publish the watermark only when step interval has passed.
```

Comment:
As discussed offline with @KeranYang, I have tested the kafka e2e test after removing the watermark config but e2e was still passing, so there maybe some deployment configuration or sendMessage process might require update, I'll sync up with @yhl25 on that.

Note: For http source it's working as expected!!

Reply:
Agree. in http we explicitly pick one of the pod for sending messages. Might need to find an equivalent for kafka.

Reply:
Verified it by adding logs, I can see it's also publishing to only one of the partition
```
2023/12/15 06:12:04 message sent, partition: 0  offset:  0
2023/12/15 06:12:04 message sent, partition: 0  offset:  1
2023/12/15 06:12:04 message sent, partition: 0  offset:  2
2023/12/15 06:12:04 message sent, partition: 0  offset:  3
2023/12/15 06:12:04 message sent, partition: 0  offset:  4
2023/12/15 06:12:04 message sent, partition: 0  offset:  5
2023/12/15 06:12:04 message sent, partition: 0  offset:  6
2023/12/15 06:12:04 message sent, partition: 0  offset:  7
2023/12/15 06:12:04 message sent, partition: 0  offset:  8
2023/12/15 06:12:04 message sent, partition: 0  offset:  9
2023/12/15 06:12:04 message sent, partition: 0  offset:  10
2023/12/15 06:12:04 message sent, partition: 0  offset:  11
2023/12/15 06:12:04 message sent, partition: 0  offset:  12
2023/12/15 06:12:04 message sent, partition: 0  offset:  13
2023/12/15 06:12:04 message sent, partition: 0  offset:  14
2023/12/15 06:12:04 message sent, partition: 0  offset:  15
2023/12/15 06:12:04 message sent, partition: 0  offset:  16
2023/12/15 06:12:04 message sent, partition: 0  offset:  17
2023/12/15 06:12:04 message sent, partition: 0  offset:  18
2023/12/15 06:12:04 message sent, partition: 0  offset:  19
2023/12/15 06:12:04 message sent, partition: 0  offset:  20
2023/12/15 06:12:04 message sent, partition: 0  offset:  21
2023/12/15 06:12:04 message sent, partition: 0  offset:  22
2023/12/15 06:12:04 message sent, partition: 0  offset:  23
2023/12/15 06:12:04 message sent, partition: 0  offset:  24
2023/12/15 06:12:04 message sent, partition: 0  offset:  25
2023/12/15 06:12:05 message sent, partition: 0  offset:  26
....
```

Reply:
Nice, did we figure out what the issue was?

Reply:
By default it's sending data to one partition, also there was one issue that test was published data to future, that i have fixed.

Reply:
Cool, I added some minor comments. We can merge after they get fixed.